### PR TITLE
Submission of demo about Cloudprober

### PR DESCRIPTION
**Pull request by altaired**
# Monitoring availability using Cloudprobe and Prometheus

## Members

Martijn Atema, atema@kth.se, GitHub: Atema
Simon Persson, siper@kth.se, GitHub: altaired

## Submission
We've now completed the demo, it consists of a 3:00 minute youtube video that can be found [here](https://youtu.be/qjzH51mBAO8) and an accompanying github repository [here](https://github.com/Atema/Cloudprober-Demo) that includes written instructions as well as all the configuration files used in the demo. With it we aim to fulfill the following criterias:

|                                             | Yes | No | Remarkable |
|-------------------------------------------- | ----|----|-------------|
|The demonstration is clearly motivated (why it matters for Devops?) | 🔥 Yes | No | Relates to a hard problem |
|The demonstration is difficult to do | 🔥 Yes | No | Relies on a non trivial infrastructure |
|The demonstration is original | Yes | No | 🔥 The are less than 10 demos on the topic on Internet |
|The video is sublime (eg visually appealing) | 🔥 Yes | No | Excellent narrative |
|The video contains an [easter egg](https://github.com/OrkoHunter/python-easter-eggs) | Yes | No | 🔥 Related to the demo |
|There is a code repo to run the demo  | Yes | No | 🔥 Code repo with a solid readme |
|The video must contain subtitles which are clear and in proper English | Yes | No | 🔥 Clearly understandable voice over |
|The video includes a take-home message | Yes | No | 🔥 Actionable takeaway |

### CODE REVIEW

**Comment by iriediese** [comment]
Hey, great video. Style looks good, subtitles are accurate, animations are where they should be, easter eggs are there, and it pretty much ticks all checkboxes.
I don't see much to improve, unfortunately. You could maybe update the description to make it slightly longer and more elaborate, but that's just nitpicking at this point.

**Comment by ysndr** [comment]
Hej

I really liked the video!
Amazing presentation of the configuration files! Also thank you for providing the files in a Repository.
I haven't heard of prometheus and cloudprober before - that's a good find.
The youtube subtitles seem to be accurate. 
As an improvement, it would be interesting to know that/if it can be run on other platforms, i.e. docker-compose or is bound to kubernetes and the google cloud.

**Comment by christian-stj** [comment]
Here's some feedback from me:

Strengths:
- Excellent video and animations, I particularly liked the borders around files and terminal as it centralizes content in the video. 
- Great content, clear and easy to follow
- Good narration

Improvements
- I think the voice recording is a bit muddy (what microphone did you use?) and the feel of the quality of the video would greatly improve by using a better microphone for a better voice quality but I understand that re-recording it is too much work.

**Comment by Adamih** [comment]
Overall:
* Incredible visual presentation in all parts of the presentation, and the visuals are appropriate. 
* In most parts it's well-paced and easy to follow. 
* Low sound & sound quality. This is hard to fix, but recording in a closet or car is a nice trick. 
* No issues with grammar/spelling in the captions. Small thing but very nice. 

Good parts:
* Good problem description.
* Fantastic animations and graphics. Amazing transitions. Very illustrative and helps with engagement. 
* Skipping boilerplate setup is nice.
* Assumes some prior technical knowledge of google cloud. This is good and keeps the demo topic more focused. 
* Configuration code highlighting is fantastic. 
* Memes. Not too many, not too few. Nice. 
* The Prometheus demo is welcoming. It's good to be able to see the end product. 

Parts that can be improved:
* Hard to follow the introduction. Use more explaining grammar or speak slower. This is not a problem in any other part. Is this because of the hard 3-minute limit?
* Rename video

**Comment by kittytinythai** [comment]
## Feedback
Overall an interesting topic you have chosen to talk about! I have listed the strengths and some suggestions for improvement below. Good job! 👍 
### Strengths
- Nice justification of the topic relevance
- Clear animations to understand different parts of a configuration files
- Good subtitles
- Visually appealing
- Good use of memes 😆 
### Suggestions for improvement
- Add a link to the used repo in the video description
- I think it will be less confusion for the audience if the video began with a short sentence that describes what the video will demonstrate
- Update the title of the video, make it more descriptive and maybe include the main use of Cloudprober as you have explained in the video

**Comment by majate** [comment]
Great demo of high quality! The highlights of relevant parts of the screens make it easy to follow along and understand what to look at. The demo is visually appealing with nice animations, have a clear content, is well motivated and has a good pace. There is not much that needs to be improved, but here are some potential points of improvement:

- The beginning of the video is a bit rushed whit the video jumping straight into the motivation of the video. As already mentioned by others, this problem may be solved by including a short introduction describing what the video will be about. If you want to keep the structure of having the introduction after giving the motivation, another way to reduce the rushed feeling could be to simply slow down the first sentence, add a short paus before starting to talk, or adding a simple "Hello" to the beginning, giving some time for the viewers to start listening.
- In the Cloudprober configuration, why do we need the two probes that probe the KTH webpage and DNS? In what way is that important for our application? I would have appreciated a motivation of that.
- I noticed that you have included a link to a repo which contains much more detailed explanations of the whole configuration. Since not everyone looks in the description box, it might be good to mention in the video that we can find more explanations and details in the description. If you don't want to spend time on mentioning it, you could instead add a text message on the screen.
